src/Database/Driver/Postgres.php

```diff
@@ -129,4 +129,29 @@ public function supportsDynamicConstraints()
     {
         return true;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function calculateColumnMap(Query $query)
+    {
+        $map = [];
+        $defaultAlias = $query->getRepository()->getAlias();
```

Comment:
`Cake\Database\Query` doesn't have a `getRepository()` method and since the driver is in database package it can't depend on `Cake\ORM\Query`.

In general truncating the identifiers after they have been already set in query is the wrong approach. The length check should be done when the identifer/alias are being generated/set.

`Cake\ORM\Query::aliasField()` is where the `Foo_bar` style identifiers are generated. So there a check using the driver should be done to validate the identifier lengh.

Reply:
Hey @ADmad thanks reviewing again. as you've asked I've updated to the pull request again.
Would you check again if this is going in the right direction (truncate when set via `aliasField`) modified `_calculateColumnMap()` to not rely on query field alias name for mapping to entity property.

It's not checking for the driver yet. can you give me a hint on how to get from `Cake\Datasource\QueryTrait` ? I was not able to locate `Cake\ORM\Query::aliasField()`

Reply:
ok, truncating in when building the query heavily affects test suite as Cake relies on these array keys! i would still go the route @markstory suggested and I've tried to approach previously.